### PR TITLE
mixed precision: time_manager reals to r8 and clean up

### PR DESCRIPTION
**Pull request by rem1776**
**Description**
replaces default reals and `double precision` values with explicit r8_kinds.

i created wrappers for get_cal_time and real_to_time_type so that they convert passed in r4's to r8 before calling the full routines. This makes all the real calculations done in r8_kind instead of default real precision. 

I also cleaned up a lot of the old html comments since some were left in this module and made a small update to the test.

**How Has This Been Tested?**
oneapi 23.0, amd 

**Checklist:**
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules
- [x] New check tests, if applicable, are included
- [x] `make distcheck` passes

### CODE REVIEW

File: time_manager/time_manager.F90 [hunk 1]
```diff
@@ -1147,73 +865,65 @@ subroutine time_assignment(time1, time2)
...
   if (.not. set_time_private(seconds,days,ticks,t,err_msg_local)) then
-    if (error_handler('function real_to_time_type',err_msg_local,err_msg)) then
+    if (error_handler('function real8_to_time_type',err_msg_local,err_msg)) then
       return
     endif
   endif
-end function real_to_time_type
+end function real8_to_time_type
+
+function real4_to_time_type(x, err_msg) result(t)
```
Comment (mlee03) [inline time_manager/time_manager.F90:910]: would this function be needed if all reals are r8_kind in time_manager?

File: time_manager/time_manager.F90 [hunk 2]
```diff
@@ -1229,51 +939,28 @@ function safe_rtoi(rval,mode) result(ival)
...
-d = time%days*dseconds_per_day*dticks_per_second + dble(time%seconds)*dticks_per_second + dble(time%ticks)
-div = d/dble(n)
+dseconds_per_day  = real(seconds_per_day, r8_kind)
+dticks_per_second = real(ticks_per_second, r8_kind)
+d = time%days*dseconds_per_day*dticks_per_second + real(time%seconds, r8_kind)*dticks_per_second + &
+    real(time%ticks, r8_kind)
+div = d/real(n, r8_kind)
 
 days = int(div/(dseconds_per_day*dticks_per_second))
 seconds = int(div/dticks_per_second - days*dseconds_per_day)
```
Comment (mlee03) [inline time_manager/time_manager.F90:962]: real(days,r8_kind)

File: time_manager/time_manager.F90 [hunk 3]
```diff
@@ -2528,7 +2049,7 @@ function increment_date_private(Time, years, months, days, hours, minutes, secon
    cmonth = cmonth + months
 
  ! Adjust year and month number when cmonth falls outside the range 1 to 12
-   cyear = cyear + floor((cmonth-1)/12.)
+   cyear = cyear + floor((cmonth-1)/12.0_r8_kind)
```
Comment (mlee03) [inline time_manager/time_manager.F90:2052]: real(cmonth-1,r8_kind)

File: time_manager/get_cal_time.F90 [hunk 4]
```diff
@@ -313,34 +322,50 @@ function get_cal_time(time_increment, units, calendar, permit_calendar_conversio
   increment_days = floor(dt/86400)
   increment_seconds = int(dt - increment_days*86400)
 else
```
Comment (mlee03) [inline time_manager/get_cal_time.F90:324]: lines 292 -323, time_increment and dt are reals doing math with ints

File: time_manager/get_cal_time.F90 [hunk 5]
```diff
@@ -313,34 +322,50 @@ function get_cal_time(time_increment, units, calendar, permit_calendar_conversio
...
+    get_calendar_time = base_time + set_time(increment_seconds, increment_days)
 endif
 
-end function get_cal_time
+end function get_calendar_time
+
 !------------------------------------------------------------------------
+
+!> For mixed precision support, just casts to passed in increment to r8
+function get_calendar_time_wrap(time_increment, units, calendar, permit_calendar_conversion)
```
Comment (mlee03) [inline time_manager/get_cal_time.F90:356]: wondering if this function is needed

**Comment by mlee03** [comment]
@rem1776 could you also add mixed precision to the pr title?  :)

File: time_manager/get_cal_time.F90 (hunk 5 above)
Comment (rem1776) [reply time_manager/get_cal_time.F90:356]: It was needed to compile with current mixed mode so I think it'll be needed

File: time_manager/time_manager.F90 (hunk 1 above)
Comment (rem1776) [reply time_manager/time_manager.F90:910]: Yeah it's for external routines, if a module is compiled with r4 it'll call this to get a new time type. It gets converted/rounded anyway by `safe_rtoi`

File: time_manager/get_cal_time.F90 (hunk 5 above)
Comment (mlee03) [reply time_manager/get_cal_time.F90:356]: do you think using class(*) for time_increment would work?  Or is that going to make things messier..

File: time_manager/time_manager.F90 [hunk 6]
```diff
@@ -930,139 +754,76 @@ function time_minus(time1, time2)
...
+type(time_type), intent(in) :: time1 !< a time interval (dividend)
+type(time_type), intent(in) :: time2 !< a time interval (divisor)
+real(r8_kind)               :: d1, d2
 
 if(.not.module_is_initialized) call time_manager_init
 
 ! Convert time intervals to floating point days; risky for general performance?
-d1 = time1%days * dble(seconds_per_day) + dble(time1%seconds) + time1%ticks/dble(ticks_per_second)
-d2 = time2%days * dble(seconds_per_day) + dble(time2%seconds) + time2%ticks/dble(ticks_per_second)
+d1 = time1%days * real(seconds_per_day, r8_kind) + real(time1%seconds, r8_kind) &
```
[the remark this answers is not in the document]
Comment (mlee03) [reply time_manager/time_manager.F90]: @rem1776 this change :)

File: time_manager/time_manager.F90 [hunk 7]
```diff
@@ -930,139 +754,76 @@ function time_minus(time1, time2)
...
+     + time1%ticks/real(ticks_per_second, r8_kind)
+d2 = time2%days * real(seconds_per_day, r8_kind) + real(time2%seconds, r8_kind) &
```
[the remark this answers is not in the document]
Comment (mlee03) [reply time_manager/time_manager.F90]: @rem1776 this one too

File: time_manager/time_manager.F90 [hunk 8]
```diff
@@ -1229,51 +939,28 @@ function safe_rtoi(rval,mode) result(ival)
...
 logical :: ltmp
 
 ! Convert time interval to floating point days; risky for general performance?
-dseconds_per_day  = dble(seconds_per_day)
-dticks_per_second = dble(ticks_per_second)
-d = time%days*dseconds_per_day*dticks_per_second + dble(time%seconds)*dticks_per_second + dble(time%ticks)
-div = d/dble(n)
+dseconds_per_day  = real(seconds_per_day, r8_kind)
+dticks_per_second = real(ticks_per_second, r8_kind)
+d = time%days*dseconds_per_day*dticks_per_second + real(time%seconds, r8_kind)*dticks_per_second + &
```
[the remark this answers is not in the document]
Comment (mlee03) [reply time_manager/time_manager.F90]: @rem1776 this one too

File: time_manager/time_manager.F90 (hunk 2 above)
Comment (mlee03) [reply time_manager/time_manager.F90]: and here

**Comment by mlee03** [comment]
@rem1776 ignore recent comments, did not see that they were outdated!

**Review by mlee03 — APPROVED** [review]

**Comment by mlee03** [comment]
@J-Lentz @uramirez @mcallic2 don't forget to review this one

**Review by mcallic2 — APPROVED** [review]

**Review by J-Lentz — APPROVED** [review]

**Pull request merged**